File: 7-kyu/graceful-tipping/python/solution.py
from math import ceil, floor, log10

def graceful_tipping(bill):
    tip = bill * 0.15
    total = ceil(tip + bill)
    if total < 10:
        return total
    mult = floor(log10(total)) - 1
    mod = 5 * (10 ** mult)
    diff = 0
    while ((total + diff) % mod) != 0:
        diff += 1
    return total + diff

File: 7-kyu/graceful-tipping/javascript/solution.js
const gracefulTipping = (bill) =>{
  let grandTotal = Math.ceil(bill * 1.15);
  console.log(grandTotal);
  if (grandTotal < 10) {
    return grandTotal;
  }
  const powerOfTen = Math.floor(Math.log10(grandTotal)) - 1;
  while (grandTotal % (5 * Math.pow(10, powerOfTen)) !== 0) {
    grandTotal++;
  }
  return grandTotal;
};
